### PR TITLE
Update GitHub Actions workflows

### PR DESCRIPTION
**Pull request by jawnsy**
* Reduce permissions for GITHUB_TOKEN issued during build
* Remove "go get" for downloading dependencies, as the build
  step will do this automatically for Go module projects
* Upgrade minimum Go version from 1.13 to 1.16
* Fix phony targets in Makefile (.PHONEY -> [.PHONY](https://www.gnu.org/software/make/manual/html_node/Phony-Targets.html))

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -16,30 +30,20 @@ jobs:
     - name: Set up Go 1.x
       uses: actions/setup-go@v2
       with:
-        go-version: ^1.13
+        go-version: ^1.14
```
Comment (ryancurrah) [inline .github/workflows/go.yml:33]: Could change this to `^1.16`.

Comment (ryancurrah) [reply .github/workflows/go.yml:33]:
```suggestion
        go-version: ^1.16
```

File: Makefile [hunk 2]
```diff
@@ -1,42 +1,42 @@
 current_dir = $(shell pwd)
 
-.PHONEY: lint
+.PHONY: lint
```
Comment (ryancurrah) [inline Makefile:3]: Good catch! 🤦

**Review by ryancurrah — APPROVED** [review]
One suggestion otherwise looks good.

File: .github/workflows/go.yml (hunk 1 above)
Comment (jawnsy) [reply .github/workflows/go.yml:33]: @ryancurrah Should I bump the version in go.mod to 1.16 as well?

Comment (ryancurrah) [reply .github/workflows/go.yml]: Ah sorry I didn't see this. It doesn't matter to much.

**Pull request merged**